lib/PN5180/PN5180.cpp

```
@@ -1,710 +1,751 @@
+// the TFT (SPI3_HOST) on the S3-DevKitC. Two drivers arbitrating one SPI
+// peripheral corrupts transactions (garbage transceiver states) and can hold
+// the bus indefinitely, starving the scan task into a task_wdt reboot. Boards
+// whose TFT owns SPI2 instead (S3-Zero) override PN5180_SPI_BUS in build flags.
+#ifndef PN5180_SPI_BUS
+  #if CONFIG_IDF_TARGET_ESP32S3
+    #define PN5180_SPI_BUS FSPI  // bus 0 = SPI2 — matches the FSPI pin naming in BoardPins.h
+  #else
+    #define PN5180_SPI_BUS HSPI  // SPI2 on classic ESP32
+  #endif
+#endif
+static SPIClass pn5180_spi(PN5180_SPI_BUS);
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)